Source/CarthageKit/Xcode.swift

```diff
@@ -355,6 +357,65 @@ private func mergeExecutables(_ executableURLs: [URL], _ outputURL: URL) -> Sign
 		.then(SignalProducer<(), CarthageError>.empty)
 }
 
+private func findSwiftHeaderFiles(fromExecutableURLs executableURLs: [URL]) -> [URL] {
+	return executableURLs.map { 
+		let executableName = $0.lastPathComponent
+		return $0.deletingLastPathComponent().appendingPathComponent("Headers/\(executableName)-Swift.h") 
+	}
+}
+
+private func findSwiftHeaderOutputURL(fromURL url: URL) -> URL {
+	let executableName = url.lastPathComponent
+	
+	return url.deletingLastPathComponent().appendingPathComponent("Headers/\(executableName)-Swift.h")
+}
+
+private func mergeSwiftHeaderFiles(_ executableURLs: [URL], _ executableOutputURL: URL) -> SignalProducer<(), CarthageError> {
+	precondition(executableOutputURL.isFileURL)
+	precondition(executableURLs.count > 1)
+
+	let headerURLs = findSwiftHeaderFiles(fromExecutableURLs: executableURLs)
+	let outputURL = findSwiftHeaderOutputURL(fromURL: executableOutputURL)
+    let conditionalPrefix = "#if 0\n#elif (defined(__x86_64__) && __x86_64__) || (defined(__i386__) && __i386__)\n"
+    let conditionalSuffix = "\n#endif"
+    
+    guard let conditionalPrefixContents = conditionalPrefix.data(using: .utf8) else { return .empty }
+    guard let conditionalSuffixContents = conditionalSuffix.data(using: .utf8) else { return .empty }
+	
+	var fileContents = Data()
+
+	for url in headerURLs {
+		guard let contents = FileManager.default.contents(atPath: url.path) else { continue }
+
+        // Need to work around an inconsistency in the way that the Swift
```

Comment:
Created radar 48743840 to try to get this behavior to be consistent across platforms.